### PR TITLE
Implemented functionality to the channels section of Inventory channels page (#258)

### PR DESCRIPTION
**Pull request by amansinghbais**
### Related Issues
 

 Related Issue #258

 ### Short Description and Why It's Useful
 
Added functionality to the channels section of the inventory channels page.

 ### Screenshots of Visual Changes before/after (If There Are Any)
 

 **IMPORTANT NOTICE** - Remember to add changelog entry

 ### Contribution and Currently Important Rules Acceptance
 

 - [x] I read and followed [contribution rules](https://github.com/hotwax/threshold-management#contribution-guideline)

### CODE REVIEW

File: src/components/LinkThresholdFacilitiesToGroupModal.vue [hunk 1]
```diff
@@ -11,45 +11,95 @@
...
+onMounted(() => {
+  selectedFacilityId.value = props.selectedConfigFacilityId?.facilityId ? JSON.parse(JSON.stringify(props.selectedConfigFacilityId.facilityId)) : '';
+})
+
+function closeModal() {
+  modalController.dismiss({ dismissed: true });
+}
+
+async function saveFacility() {
+  if(!selectedFacilityId.value) {
```
Comment (ymaheshwari1) [inline src/components/LinkThresholdFacilitiesToGroupModal.vue:63]: Add check to only update when there is some change in facilityId

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/LinkThresholdFacilitiesToGroupModal.vue (hunk 1 above)
Comment (amansinghbais) [reply src/components/LinkThresholdFacilitiesToGroupModal.vue:63]: Improved code to disable save button until config facility updated.

File: src/components/LinkFacilitiesToGroupModal.vue [hunk 2]
```diff
@@ -11,45 +11,136 @@
...
 import { translate } from '@/i18n'
-</script>
\ No newline at end of file
+import { defineProps, onMounted, ref } from "vue";
+import { useStore } from "vuex";
+import { DateTime } from "luxon";
+import { ChannelService } from "@/services/ChannelService";
+import { UtilService } from "@/services/UtilService";
+import { showToast } from "@/utils";
+import { hasError } from "@/utils";
```
Comment (ymaheshwari1) [inline src/components/LinkFacilitiesToGroupModal.vue:48]: Use single import.

File: src/components/LinkFacilitiesToGroupModal.vue [hunk 3]
```diff
@@ -11,45 +11,136 @@
...
+}
+
+async function fetchFacilities () {
+  facilities.value = []
+
+  try {
+    const params = {
+      productStoreId: store.state.user.currentEComStore.productStoreId,
+      facilityName: queryString.value,
+      facilityName_op: "contains",
```
Comment (ymaheshwari1) [inline src/components/LinkFacilitiesToGroupModal.vue:74]: Add check to pass facilityName filter only when queryString is available.

File: src/components/LinkFacilitiesToGroupModal.vue [hunk 4]
```diff
@@ -11,45 +11,136 @@
...
+      facilities.value = resp.data;
+    } else {
+      throw resp.data
+    }
+  } catch (err: any) {
+    logger.error(err)
+  }
+}
+
+function isFacilitySelected(facilityId: any) {
```
Comment (ymaheshwari1) [inline src/components/LinkFacilitiesToGroupModal.vue:90]:
```suggestion
function isFacilitySelected(facilityId: string) {
```

File: src/components/LinkFacilitiesToGroupModal.vue [hunk 5]
```diff
@@ -11,45 +11,136 @@
...
+  return selectedFacilityValues.value.some((facility: any) => facility.facilityId === facilityId)
+}
+
+function updateSelectedFacilities(id: string) {
+  const facility = isFacilitySelected(id)
+
+  if(facility) {
```
Comment (ymaheshwari1) [inline src/components/LinkFacilitiesToGroupModal.vue:97]:
```suggestion
  if(isFacilitySelected(id)) {
```

File: src/components/CreateGroupModal.vue [hunk 6]
```diff
@@ -13,30 +13,150 @@
...
+
+  if (formData.value.facilityGroupId.length > 20) {
+    showToast(translate("Internal ID cannot be more than 20 characters."))
+    return
+  }
+
+  // In case the user does not lose focus from the facility name input
+  // and click on create the button, we need to set the internal id manually
+  if (!formData.value.facilityGroupId) {
+    formData.value.facilityGroupId = generateInternalId(formData.value.facilityGroupName)
```
Comment (ymaheshwari1) [inline src/components/CreateGroupModal.vue:87]: If user enters a facilityGroupName having more than 20 characters then facilityGroupId generated in this step will be more than 20 characters, but we have previously added a check to not create facilityGroup if Id is more than 20 characters.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/channel/mutations.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { MutationTree } from 'vuex'
+import ChannelState from './ChannelState';
+import * as types from './mutation-types'
+
+const mutations: MutationTree <ChannelState> = {
+  [types.CHANNEL_INVENTORY_CHANNELS_UPDATED] (state, payload) {
+    state.inventoryChannels = payload;
+  },
+  [types.CHANNEL_CLEARED](state) {
```
Comment (ymaheshwari1) [inline src/store/modules/channel/mutations.ts:9]: No need to have a separate mutation to clear the state we can use `CHANNEL_INVENTORY_CHANNELS_UPDATED` mutation.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Review by ymaheshwari1 — APPROVED** [review]

**Pull request merged**